packages/core/helper-plugin/src/features/StrapiApp.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+
+import { TranslationMessage } from 'types';
+
+interface Permission {
+  action: string;
+  subject: string | null;
+}
+
+interface MenuItem {
+  to: string;
+  icon: React.ComponentType;
+  intlLabel: TranslationMessage;
+  permissions?: Permission[];
+  Component?: React.ComponentType;
+}
+
+/* -------------------------------------------------------------------------------------------------
+ * Context
+ * -----------------------------------------------------------------------------------------------*/
+
+interface Plugin {
+  apis: Record<string, unknown>;
+  injectionZones: Record<string, unknown>;
+  initializer: ({ setPlugin }: { setPlugin: (pluginId: string) => void }) => null | null;
+  isReady: boolean;
+  name: string;
+  pluginId: string;
+}
+
+interface StrapiAppSettingLink {
+  id: string;
+  to: string;
+  intlLabel: TranslationMessage;
+  Component: React.ComponentType;
+  permissions: Permission[] | [];
+}
+
+interface StrapiAppSetting {
+  id: string;
+  intlLabel: TranslationMessage;
+  links: StrapiAppSettingLink[];
+}
```

Comment:
This is based off of:
![Screenshot 2023-09-05 at 16 13 33](https://github.com/strapi/strapi/assets/26598053/3a2f894f-f0e5-4260-9108-e320a1ea33b0)

The shape looks consistent but I'm not positive.

Reply:
I think it's fine for now, we can refine if needs be :)